common/src/main/java/org/opensearch/ml/common/model/MLGuard.java

```diff
@@ -122,17 +132,26 @@ public Boolean validateStopWordsSingleIndex(String input, String indexName, List
             searchSourceBuilder.parseXContent(queryParser);
             searchSourceBuilder.size(1); //Only need 1 doc returned, if hit.
             searchRequest = new SearchRequest().source(searchSourceBuilder).indices(indexName);
-            searchResponse = client.search(searchRequest).actionGet(1000l * 30);
             context.restore();
+            client.search(searchRequest, ActionListener.runBefore(new LatchedActionListener(ActionListener.<SearchResponse>wrap(r -> {
+                if (r == null || r.getHits() == null || r.getHits().getTotalHits() == null || r.getHits().getTotalHits().value == 0) {
+                    hitStopWords.set(true);
+                }
+            }, e -> {
+                log.error("Failed to search stop words index {}", indexName, e);
+                hitStopWords.set(true);
+            }), latch), () -> context.restore()));
         } catch (Exception e) {
             log.error("[validateStopWords] Searching stop words index failed.", e);
-            return false;
+            hitStopWords.set(true);
```

Comment:
Should we decrease count down latch here?

Reply:
Good catch, countdown here.